blockchain.com/src/adapter.ts

```diff
@@ -6,51 +6,50 @@ export type JobSpecRequest = {
   id: string
   data: Record<string, unknown>
 }
-type Callback = (statusCode: number, data: Record<string, unknown>) => void
+type JobSpecResponse = { statusCode: number; data: Record<string, unknown> }
 
 const inputParams = {
   endpoint: false,
 }
 
-const config: Config = getConfig()
-logConfig(config)
-
 // Export function to integrate with Chainlink node
-export const execute = (request: JobSpecRequest, callback: Callback): void => {
+export const execute = async (
+  request: JobSpecRequest,
+  config: Config,
+): Promise<JobSpecResponse> => {
   const validator = new Validator(request, inputParams)
-  if (validator.error) return callback(validator.error.statusCode, validator.error)
-
-  const jobRunID = validator.validated.id
-
-  const _handleResponse = (out: any): void => {
-    callback(
-      200,
-      Requester.success(jobRunID, {
-        data: { result: out },
-        result: out,
-        status: 200,
-      }),
-    )
-  }
+  if (validator.error) throw validator.error
 
-  const _handleError = (err: Error): void => callback(500, Requester.errored(jobRunID, err.message))
+  logConfig(config)
 
+  const jobRunID = validator.validated.id
   const endpoint = validator.validated.data.endpoint || DEFAULT_ENDPOINT
 
   let fn
   switch (endpoint) {
     case balance.Name: {
       const validator = new Validator(request, balance.inputParams)
-      if (validator.error) return callback(validator.error.statusCode, validator.error)
+      if (validator.error) throw validator.error
 
       fn = balance.execute(config, request, validator.validated.data)
       break
     }
     default: {
-      fn = Promise.reject(Error(`Endpoint ${endpoint} not supported.`))
-      break
+      throw Error(`Endpoint ${endpoint} not supported.`)
     }
   }
 
-  fn.then(_handleResponse).catch(_handleError)
+  const result = await fn
```

Comment:
Can we await when `balance.execute` is first called? Then store the result in the variable instead of the promise?

Reply:
Great catch! We can now that this is `async`, it was full of promises and callbacks just a minute ago ;)